SearchAndSortOnEmployee/SearchAndSortOnEmployee.java

```diff
@@ -0,0 +1,309 @@
+
+  }
+
+  static void delete() {
+    // here we will first remove the specified employee from set and rewrite the set data to employee.txt file
+    try {
+      System.out.println("Enter the email id of the employee: ");
+      String emailId = sc.next();
+
+      Iterator<Employee> iterator = set.iterator();
+      while (iterator.hasNext()) { // we have used iterator in place of for loop because java throws an exception
+                                   // if we apply loop on a class and remove from the same class in it.
+        Employee employee = iterator.next();
+        if (employee.getEmail().equals(emailId)) {
+          set.remove(employee);
+          break;
+        }
+      }
+
+      String employeeFilePath = "employees.txt";
+      File employeeFile = new File(employeeFilePath);
+      FileWriter fileWriter = new FileWriter(employeeFilePath);
+      BufferedWriter bufferedWriter = new BufferedWriter(fileWriter);
+
+      for (Employee employeeObject : set) {
+        bufferedWriter.write("Name: " + employeeObject.getName() + ", Age: " + employeeObject.getAge() + ", Email: "
+            + employeeObject.getEmail()
+            + ", Date of birth: " + employeeObject.getDateOfBirth());
+        bufferedWriter.newLine();
+
+      }
+
+      bufferedWriter.close();
+      fileWriter.close();
+
+      System.out.println("Employee " + emailId + " is removed successfully.");
+      System.out.println("-------------------------------\n");
+      if (Desktop.isDesktopSupported()) { // Desktop class is used to open employee.txt file after adding data to the file
+        Desktop desktop = Desktop.getDesktop();
+        desktop.open(employeeFile);
+      }
+
+    } catch (Exception e) {
+      System.out.println(e);
+    }
+
+  }
+
+  static void fetch() { 
+    // if employee.txt file is already present then we will fetch its data and add it to set.
+    try {
+      File employeeFile = new File("employees.txt");
+      if (employeeFile.createNewFile() == false) { // employee.txt already exists
+        FileReader fileReader = new FileReader("employees.txt");
+        BufferedReader bufferedReader = new BufferedReader(fileReader);
+        String employee = "";
+        while ((employee = bufferedReader.readLine()) != null) {
+
+          String[] employeeData = employee.split(", "); // all employee's data
+
+          Employee employeeObject = new Employee();
+          employeeObject.setName(employeeData[0].split(" ")[1]);
+          employeeObject.setAge(Integer.parseInt(employeeData[1].split(" ")[1]));
+          employeeObject.setEmail(employeeData[2].split(" ")[1]);
+          employeeObject.setDate(employeeData[3].split(" ")[3]);
+
+          set.add(employeeObject);
+
+        }
+
+        bufferedReader.close();
+      }
+
+    } catch (Exception e) {
+      System.out.println(e);
+    }
+
+  }
+
+  static void searchAndSort() {
+    try {
+      System.out.println("Enter which field to search: (name, age, date of birth)");
```

Comment:
No validation if search field exists, program does not abort or throws error if non existent field is specified for searching, rectify

Reply:
Resolved